src/logic/map_objects/tribes/tribe_descr.cc

```diff
@@ -546,4 +637,242 @@ DescriptionIndex TribeDescr::add_special_ware(const std::string& warename) {
+
+	// Register who creates which entities
+	std::map<std::string, std::set<ProductionSiteDescr*>> creators;
+	auto add_creator = [&creators](const std::string& item, ProductionSiteDescr* productionsite) {
+		if (creators.count(item) != 1) {
+			creators[item] = {productionsite};
+		} else {
+			creators[item].insert(productionsite);
+		}
+	};
+
+	// Register who collects which entities
+	std::map<std::string, std::set<ProductionSiteDescr*>> collectors;
+	auto add_collector = [&collectors](
+	                        const std::string& item, ProductionSiteDescr* productionsite) {
+		if (collectors.count(item) != 1) {
+			collectors[item] = {productionsite};
+		} else {
+			collectors[item].insert(productionsite);
+		}
+	};
+
+	for (ProductionSiteDescr* prod : productionsites) {
+		// Add bobs that are created directly
+		for (const std::string& bobname : prod->created_bobs()) {
+			const CritterDescr* critter = world.get_critter_descr(bobname);
+			if (critter == nullptr) {
+				if (worker_index(bobname) == Widelands::INVALID_INDEX) {
+					throw GameDataError(
+					   "Productionsite '%s' has unknown bob '%s' in production or worker program",
+					   prod->name().c_str(), bobname.c_str());
+				}
+			}
+			add_creator(bobname, prod);
+		}
+
+		// Get attributes and bobs from transformations
+		std::set<std::string> deduced_bobs;
+		std::set<std::string> deduced_immovables;
+		// Remember where we walked in case of circular dependencies
+		std::set<DescriptionIndex> walked_world_immovables;
+		std::set<DescriptionIndex> walked_tribe_immovables;
+
+		for (const auto& attribinfo : prod->created_attributes()) {
+			const MapObjectType mapobjecttype = attribinfo.first;
+			const MapObjectDescr::AttributeIndex attribute_id = attribinfo.second;
+			if (mapobjecttype != MapObjectType::IMMOVABLE) {
+				continue;
+			}
+			for (DescriptionIndex i = 0; i < world_immovables.size(); ++i) {
+				const ImmovableDescr& immovable_descr = world_immovables.get(i);
+				if (immovable_descr.has_attribute(attribute_id)) {
+					walk_world_immovables(i, world, &walked_world_immovables, needed_attributes,
+					                      &deduced_immovables, &deduced_bobs);
+					if (needed_attributes.count(attribute_id) == 1) {
+						prod->add_created_immovable(immovable_descr.name());
+						add_creator(immovable_descr.name(), prod);
+					}
+				}
+			}
+			for (const DescriptionIndex i : immovables()) {
+				const ImmovableDescr& immovable_descr = *get_immovable_descr(i);
+				if (immovable_descr.has_attribute(attribute_id)) {
+					walk_tribe_immovables(i, *this, &walked_tribe_immovables, needed_attributes,
+					                      &deduced_immovables, &deduced_bobs);
+					if (needed_attributes.count(attribute_id) == 1) {
+						prod->add_created_immovable(immovable_descr.name());
+						add_creator(immovable_descr.name(), prod);
+					}
+				}
+			}
+
+			// We're done with this site's attributes, let's get some memory back
+			prod->clear_attributes();
+		}
```

Comment:
```suggestion
		}
		// We're done with this site's attributes, let's get some memory back
		prod->clear_attributes();
```
This fixes the segfault